### PR TITLE
[MINOR] Remove extra anonymous closure within functional transformations

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR removes extra anonymous closure within functional transformations.

For example,

``` scala
.map(item => {
  ...
})
```

which can be just simply as below:

``` scala
.map { item => 
  ...
}
```
## How was this patch tested?

Related unit tests and `sbt scalastyle`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @rxin

**Comment by rxin** [comment]
LGTM pending tests.

**Comment by HyukjinKwon** [comment]
retest this please

File: examples/src/main/scala/org/apache/spark/examples/streaming/RecoverableNetworkWordCount.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ object RecoverableNetworkWordCount {
     val lines = ssc.socketTextStream(ip, port)
     val words = lines.flatMap(_.split(" "))
     val wordCounts = words.map(x => (x, 1)).reduceByKey(_ + _)
-    wordCounts.foreachRDD((rdd: RDD[(String, Int)], time: Time) => {
+    wordCounts.foreachRDD { (rdd: RDD[(String, Int)], time: Time) =>
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/streaming/RecoverableNetworkWordCount.scala]: While we're here, can this just be `(rdd, time) =>`? and likewise in the following file, maybe others

**Comment by srowen** [comment]
Merged to master

**Comment by andrewor14** [comment]
Maybe we should add a scalastyle rule for this

**Comment by HyukjinKwon** [comment]
@andrewor14 Hm.. I tried to add a rule but I realised (or I think) it cannot find perfectly by a static way. One of the problems I met was below:

For example, there is a class `GraphOps` and at line 74,

```scala
graph.aggregateMessages(ctx => { ctx.sendToSrc(1); ctx.sendToDst(1) }, _ + _,
    TripletFields.None)
}
```

In this case, this should not be the error case since `(` and `)` exists for other arguments ` _ + _` and `TripletFields.None`.

In order to check this, it should be able to check `,` after `ctx => { ctx.sendToSrc(1); ctx.sendToDst(1) }`. In order to skip `ctx => { ctx.sendToSrc(1); ctx.sendToDst(1) }`, we need to skip the range between `{` and `}` which might have nested `{`, `}` and `,`. AFAIK, it is impossible for regex in theory but it looks possible in practical by recursive regular expression but it looks scalastyle does not support this.

Maybe i am not smart enough but this was just my conclusion.

**Comment by HyukjinKwon** [comment]
@andrewor14 @srowen Do you mind If I open another PR for some more corrections?

While trying to test regular expressions, I found some more things to correct about this and about @srowen comment, which are pretty many.

**Comment by srowen** [comment]
If there's a built-in scalastyle rule for this, let's enable it, but otherwise not sure it's worth building custom code to enforce it. If you've found ways to look for more instances of this and found more classes of this problem, OK open a new PR.

**Pull request closed without merging**